presto-connector-tests/src/main/java/com/facebook/presto/connector/unittest/MetadataSchemaTest.java

```diff
@@ -0,0 +1,73 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.unittest;
+
+import com.facebook.presto.connector.meta.RequiredFeatures;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.testing.TestingConnectorSession;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.util.Map;
+
+import static com.facebook.presto.connector.meta.ConnectorFeature.CREATE_SCHEMA;
+import static com.facebook.presto.connector.meta.ConnectorFeature.DROP_SCHEMA;
+import static com.facebook.presto.connector.meta.ConnectorFeature.RENAME_SCHEMA;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@RequiredFeatures({CREATE_SCHEMA, DROP_SCHEMA})
+public interface MetadataSchemaTest
+        extends BaseMetadataTest
+{
+    Map<String, Object> getTableProperties();
+
+    @Test
+    default void testCreateDropSchema()
+            throws Exception
+    {
+        ConnectorSession session = new TestingConnectorSession(ImmutableList.of());
+        String schemaName = "testCreateSchema";
+
+        withMetadata(
+                ImmutableList.of(
+                        metadata -> metadata.createSchema(session, schemaName, ImmutableMap.of()),
+                        metadata -> assertEquals(getOnlyElement(metadata.listSchemaNames(session)), schemaName),
+                        metadata -> assertEquals(metadata.listTables(session, schemaName).size(), 0),
+                        metadata -> assertEquals(metadata.listTables(session, null).size(), 0),
+                        metadata -> metadata.dropSchema(session, schemaName)));
+    }
+
+    @Test
+    @RequiredFeatures({RENAME_SCHEMA})
+    default void testRenameSchema()
+            throws Exception
+    {
+        ConnectorSession session = new TestingConnectorSession(ImmutableList.of());
+        String initialSchemaName = "testRenameSchemaInitial";
+        String renamedSchemaName = "testRenameSchemaRenamed";
+
+        withMetadata(
+                ImmutableList.of(
+                        metadata -> metadata.createSchema(session, initialSchemaName, ImmutableMap.of()),
+                        metadata -> metadata.renameSchema(session, initialSchemaName, renamedSchemaName),
+                        metadata -> assertEquals(getOnlyElement(metadata.listSchemaNames(session)), renamedSchemaName),
+                        metadata -> assertEquals(metadata.listTables(session, renamedSchemaName).size(), 0),
+                        metadata -> assertEquals(metadata.listTables(session, null).size(), 0),
+                        metadata -> metadata.dropSchema(session, renamedSchemaName)));
```

Comment:
Why do you have one consumer per call?

Reply:
Also covered here:

https://github.com/prestodb/presto/pull/8723#discussion_r134818591